system-tests/test/deposit-redemption.test.ts

```diff
@@ -0,0 +1,333 @@
+        true,
+        [depositUtxo],
+        [deposit]
+      ))
+
+      console.log(`
+        Deposit swept on Bitcoin chain:
+        - Transaction hash: ${sweepUtxo.transactionHash}
+      `)
+
+      // Unlike in the deposit transaction case, we must wait for the sweep
+      // transaction to have an enough number of confirmations. This is
+      // because the bridge performs the SPV proof of that transaction.
+      await waitTransactionConfirmed(electrumClient, sweepUtxo.transactionHash)
+
+      await fakeRelayDifficulty(
+        systemTestsContext,
+        electrumClient,
+        sweepUtxo.transactionHash
+      )
+
+      // TODO: Consider fetching the current wallet main UTXO and passing it
+      //       here. This will allow running this test scenario multiple
+      //       times for the same wallet.
+      await TBTC.proveDepositSweep(
+        sweepUtxo.transactionHash,
+        // This is the first sweep of the given wallet so there is no main UTXO.
+        {
+          // The function expects an unprefixed hash.
+          transactionHash: constants.HashZero.substring(2),
+          outputIndex: 0,
+          value: BigNumber.from(0),
+        },
+        maintainerBridgeHandle,
+        electrumClient
+      )
+
+      console.log(`
+        Deposit sweep proved on the bridge
+      `)
+    })
+
+    it("should broadcast the sweep transaction on the Bitcoin network", async () => {
+      expect(
+        (await electrumClient.getRawTransaction(sweepUtxo.transactionHash))
+          .transactionHex.length
+      ).to.be.greaterThan(0)
+    })
+
+    it("should sweep the deposit on the bridge", async () => {
+      const { sweptAt } = await getDepositFromBridge(
+        systemTestsContext,
+        depositUtxo
+      )
+      expect(sweptAt).to.be.greaterThan(0)
+    })
+
+    it("should increase depositor's balance in the bank", async () => {
+      const { treasuryFee } = await getDepositFromBridge(
+        systemTestsContext,
+        depositUtxo
+      )
+
+      const expectedBalance = depositAmount
+        .sub(treasuryFee)
+        .sub(depositSweepTxFee)
+
+      const actualBalance = await bank.balanceOf(
+        systemTestsContext.depositor.address
+      )
+
+      expect(actualBalance).to.be.equal(expectedBalance)
+    })
+  })
+
+  context("when redemption is requested", () => {
+    let requestedAmount: BigNumber
+    let redeemerOutputScript: string
+
+    before("request the redemption", async () => {
+      // Redeem the full depositor's balance.
+      requestedAmount = await bank.balanceOf(
+        systemTestsContext.depositor.address
+      )
+
+      // Allow the bridge to take the redeemed bank balance.
+      await bank
+        .connect(systemTestsContext.depositor)
+        .approveBalance(bridgeAddress, requestedAmount)
+
+      // Request redemption to depositor's address.
+      redeemerOutputScript = `0014${computeHash160(
```

Comment:
Side question: Do we expect in `tbtc-v2.ts` that a redeemer will provide an output script as a target for redemption transaction? Wouldn't it be too complex for the user? Shouldn't we require a bitcoin address?

Reply:
Requiring the output script is more universal. Of course, the dApp will expect the address and derive the output script under the hood. This is a simple operation. We can of course expose a conversion function from the `tbtc-v2.ts` library in the future if needed.

Reply:
> This is a simple operation

Yeah 😉 

![beautiful mind](https://media.giphy.com/media/ohdY5OaQmUmVW/giphy.gif)